### ISSUE TITLE
sdk_test

### ISSUE THREAD

**Issue by accqaz** [wasinsangdam/LeNet-5#1]
Hello! Excuse me, when I was testing the software with the Vivado 2018.3 SDK, I reported an error in the “arm/main.cc” the “iostream” header file in the “arm/main.cc” file reported an error that the file could not be found, and **delete** and **new** appeared as "undefined". I would like to ask which version of Vivado is used for on-board verification and how to solve it？Thank you very much！

**Comment by wasinsangdam** [comment]
Hi, Thank you for question!

I used Vivado 2019.1 and Xilinx SDK 2019.1, but your problem doesn't seem to be a version problem.

I think the cause of the problem is that the target software language is `C` and not `C++`.

When you first set up your SDK project, set the target software language to `C++`.

==================================================================================


![image](https://github.com/wasinsangdam/LeNet-5/assets/116902937/11cb3fc7-6ce5-488a-ace6-6cd570d8415e)

==================================================================================

Additionally, since I saved the input binary files to the SD card, you need to additionally set up the BSP(Board Support Package).

In `Project Explorer`, Right click `*_bsp` -> `Board Support Package Settings` -> Check `xilffs`

==================================================================================

![image](https://github.com/wasinsangdam/LeNet-5/assets/116902937/e9f86c99-cca2-429a-ba26-bc6716a04699)

![image](https://github.com/wasinsangdam/LeNet-5/assets/116902937/494a904c-1ab7-4e1c-bc52-73f9ed98a671)

==================================================================================

I recommend you to watch this video. [Youtube Link](https://youtu.be/i_zWX0mBt1k?si=2H85H2kw0G7JMHKL)

It will be helpful to understand how get data via SD card to PS(Processing System).

If you have any questions, feel free to ask more anytime.

I hope my answer helps you.

**Comment by accqaz** [comment]
Hello!


I'm sorry to bother you again. The problem I consulted this morning has been successfully solved. However，I am sorry that I have another problem. That is in SDK test. I'm so sorry to ask you how to write and debug the test code of the SDK software? 




At present, I have connected the development board and stored input in the SD card in advance, but there is no output from the terminal when I run it. Could you help me? Thank you very much?
I wish you a pleasant day!


------------------ 原始邮件 ------------------
发件人:                                                                                                                        "wasinsangdam/LeNet-5"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年8月22日(星期二) 中午1:46
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [wasinsangdam/LeNet-5] sdk_test (Issue #1)





 
Hi, Thank you for question!
 
I used Vivado 2019.1 and Xilinx SDK 2019.1, but your problem doesn't seem to be a version problem.
 
I think the cause of the problem is that the target software language is C and not C++.
 
When you first set up your SDK project, set the target software language to C++.
 
==================================================================================
 

 
==================================================================================
 
Additionally, since I saved the input binary files to the SD card, you need to additionally set up the BSP(Board Support Package).
 
In Project Explorer, Right click *_bsp -&gt; Board Support Package Settings -&gt; Check xilffs
 
==================================================================================
 

 

 
==================================================================================
 
I recommend you to watch this video. Youtube Link
 
It will be helpful to understand how get data via SD card to PS(Processing System).
 
If you have any questions, feel free to ask more anytime.
 
I hope my answer helps you.

**Comment by accqaz** [comment]
Hello!


I'm sorry to bother you again. The problem I consulted yesterday has been successfully solved. However，I am sorry that I have another problem. That is in SDK test：there is no output in console，no [ERROR] and no [INFO]. I'm so sorry to ask you how to write and debug the test code of the SDK software?
![1](https://github.com/wasinsangdam/LeNet-5/assets/70732383/53e4135f-ada8-42a4-870a-e86d25b18e07)
![8)3}ERJG4LH7W`4GIC}TBF0](https://github.com/wasinsangdam/LeNet-5/assets/70732383/ae41862e-634f-439d-9568-a55365d21768)

At present, I have connected the board and stored input in the SD card in advance, but there is no output from the terminal when I run it. Could you help me? Thank you very much！
I wish you a pleasant day!

**Comment by wasinsangdam** [comment]
In my case, I used `Serial Monitor` [Link](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-serial-monitor), one of `Visaul Studio Code` extensions.

It is probably possible to use `Terminal`, not `SDK Termial` in the SDK.

`Terminal` can be used as follows.
In SDK, `Window` -> `Show View` -> `Other..` -> `Terminal` -> `Terminal`
When `Terminal` window appears, click `Connect` button and set `Encoding` and `Connection type`.

Currently I don't have a board, so I'm not sure, but the `Connection type` will be `Serial`, and `Buad Rate` will be `115200`.

I recommend using a extension like `Visual Studio Code`'s `Serial Monitor` or other program `Teraterm`.

I wish you success. Thank you!

**Comment by wasinsangdam** [comment]
Additionaly, I attach my IP config and project setting.

- Zynq7 Processing System
  - Zynq Block Design  
    
    ![image](https://github.com/wasinsangdam/LeNet-5/assets/116902937/06fd928e-4868-4a87-8c8e-ccfbc5dd413a)
  - Peripheral I/O Pins
         
    ![image](https://github.com/wasinsangdam/LeNet-5/assets/116902937/914c321e-b908-4267-b3ad-0cd7c2cf4a19)
  - Clock Configuration
 
    ![image](https://github.com/wasinsangdam/LeNet-5/assets/116902937/8219eaf1-7ec1-4272-a7af-56d0cea88afb)

  - Interrupts
  
    ![image](https://github.com/wasinsangdam/LeNet-5/assets/116902937/616e19be-a8c1-4d68-a3f8-3e2e8c6c1e20)

- AXI Direct Memory Access
  
  ![image](https://github.com/wasinsangdam/LeNet-5/assets/116902937/3af5b6b2-86fd-4e0d-80b5-7e49ba9342a5)

In Vivado, `Project Manager` -> `Settings` -> `Synthesis` -> `Strategy` -> Select `Flow_AreaOptimized_high`

Hope this information helps.

**Comment by accqaz** [comment]
Thank you very much for your patience in responding to my questions! Fortunately, there is now an output in Terminal, but there is still a problem: "terminate called without an active exception". I looked up some information but still couldn't solve this problem. Perhaps you may know the reason.
![R$_EKOY{XP)D3 QG4NJXIWU](https://github.com/wasinsangdam/LeNet-5/assets/70732383/440faffe-f061-4ae6-982b-601b6d4880c8)
In addition, the c++ code in the file `cplusplus/etc/gen_table`, I did not add them to the HLS design at runtime, would it affect the SDK test?
Wish you happiness everyday!

**Comment by wasinsangdam** [comment]
[Probelm] "terminate called without an active exception"

[Answer]
This is a problem caused by insufficient heap size.

You can increase the heap size in `lscript.ld`.

![image](https://github.com/wasinsangdam/LeNet-5/assets/116902937/9c1e6619-6823-4170-a20a-70e0d00115d9)

If you increase the heap size like me, it will work fine.

[Question]
In addition, the c++ code in the file `cplusplus/etc/gen_table`, I did not add them to the HLS design at runtime, would it affect the SDK test?

[Answer]
 `cplusplus/etc/gen_table` directory is just generate weight array from `data` directory.

I used `gen_table` directory when write the weight value in the `inc/weight.h` file.

But, in `hls`, I divided `conv1` and `conv2` weight array to use paralleism. 

I think it would not affect the SDK test.

Thank you!

**Comment by accqaz** [comment]
Thank you very, very much indeed! With your patient answers, I made it! Thank you very much for your willingness to solve some of my problems. I just started to do some FPGA-related research , so there are lots of things I do not understand. I am really touched by your patience to help me, and I have learned a lot in this process. Sincerely wish you all the best and happy every day!
Thanks so much for your help!
